### DISCUSSION TITLE
Continued maintenance!

### DISCUSSION THREAD

**Discussion by philipobenito** [thephpleague/route#339] [Announcements]
Hi, just wanted to drop an update here.

Without going in to detail, I've been dealing with health issues for the past 2-3 years, so maintenance on this package has all but stopped.

Things are improving now, and I'm in the process of updating league/container at the moment, once that is done, league/route will get the same treatment.

Watch this thread for an update in the coming weeks when I've had chance to determine an action plan.

Thanks,
Phil

**Comment by arabcoders** [comment]
Hi, Thank you for all of your hard work, hope you are getting better.

While you were away i somewhat did some improvements on how the router call the routes,

Take look at this https://github.com/arabcoders/watchstate/blob/master/src/Libs/Extends/RouterStrategy.php and https://github.com/arabcoders/watchstate/blob/master/src/Libs/Extends/ReflectionContainer.php

With this approach you are able to add the route params as method parameters, and you can ignore RequestInterface being injected automatically if the route doesn't really care for it. If you are interested i can hopefully push a PR before 6.x goes live.

This changes mostly contained in `ReflectionContainer` and the `RouterStrategy::invokeRouteCallable()` method.

One small request, i would like the `RouterStrategy::getOptionsCallable()` to receive the request object so i can also forgo `$_SERVER` check.
